### PR TITLE
Changing defaults for mitochondria mode now that we have adaptive pruning

### PR DESCRIPTION
**Pull request by meganshand**
This optimizes the defaults in mitochondria-mode for WGS mitochondria calling. It changes the `pruning-lod-threshold` in adaptive pruning and the `lod-divided-by-depth` threshold in `FilterMutectCalls`.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java [hunk 1]
```diff
@@ -29,11 +30,21 @@
...
+        final ReadThreadingAssemblerArgumentCollection rtaac = assemblerArgs;
+        final ReadThreadingAssembler assemblyEngine = rtaac.makeReadThreadingAssembler();
+        assemblyEngine.setDebug(debug);
+        assemblyEngine.setMinBaseQualityToUseInAssembly(minBaseQualityScore);
+
+        return assemblyEngine;
+    }
+
+    protected ReadThreadingAssemblerArgumentCollection getReadThreadingAssemblerArgumentCollection() {
+        return new HaplotypeCallerReadThreadingAssemblerArgumentCollection();
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java]: I would leave this `abstract` and implement it in both `HaplotypeCallerArgumentCollection` and `Mutect2ArgumentCollection`.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java [hunk 2]
```diff
@@ -29,11 +30,21 @@
     @ArgumentCollection
     public AssemblyRegionTrimmerArgumentCollection assemblyRegionTrimmerArgs = new AssemblyRegionTrimmerArgumentCollection();
 
-    protected boolean useMutectAssemblerArgumentCollection() { return false; }
+    public ReadThreadingAssembler createReadThreadingAssembler() {
+        final ReadThreadingAssemblerArgumentCollection rtaac = assemblerArgs;
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java]: Inline this variable.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/M2ArgumentCollection.java [hunk 3]
```diff
@@ -48,8 +51,20 @@
     public static final double DEFAULT_INITIAL_LOD = 2.0;
     public static final double DEFAULT_MITO_INITIAL_LOD = 0;
 
+    public static final double DEFAULT_PRUNING_LOG_ODDS_THRESHOLD = -4;
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/M2ArgumentCollection.java]: Make clear that this is the default for mitochondria.

**Comment by davidbenjamin** [comment]
@meganshand back to you.

**Comment by meganshand** [comment]
Thanks @davidbenjamin, back to you!

**Comment by davidbenjamin** [comment]
:thumbsup:

**Pull request merged**